lib/core/src/system/gpu/data/texture.rs

```diff
@@ -326,49 +331,72 @@ with_texture_format_relations!(generate_internal_format_instances []);
 
 
 
-// =====================
-// === TextureSource ===
-// =====================
+// =======================
+// === Texture sources ===
+// =======================
+
+// === Empty Texture ===
+
+/// An empty texture, created as 1x1 texture in Rgba format.
+pub struct EmptyTexture;
+
 
-/// Source of the texture. Please note that the texture will be loaded asynchronously on demand.
-#[derive(Clone,Debug)]
-pub enum TextureSource {
-    /// URL the texture should be loaded from. This source implies asynchronous loading.
-    Url(String)
+
+// === Texture from url ===
+
+/// Texture loaded from given url. It is loaded to gpu asynchronously, putting a mocked texture
+/// before
+pub struct TextureFromUrl {
+    /// An url from where the texture is loaded.
+    pub url       : String,
 }
 
-impl<S:Str> From<S> for TextureSource {
+
+impl<S:Str> From<S> for TextureFromUrl {
     fn from(s:S) -> Self {
-        Self::Url(s.into())
+        Self{url:s.into()}
     }
 }
 
 
+// === Texture from memory ===
+
+/// A texture created from slice.
+pub struct TextureFromMemory<'a,T> {
+    /// A slice from which the texture data are taken.
+    pub view   : &'a[T],
```

Comment:
I don't like the idea that the texture does not own this data. IMO data should be owned by the texture and everyone who has access to the uniform should be able to modify this data. Let's talk about it via phone.

Reply:
But I dislike idea that the FontRenderInfo know anything about uniforms etc. - and it should be able to extend msdf data, and we don't want to copy them each time the texture is created.

Reply:
Ok, the `TextureFromMemory` (and therefore `Texture` itself) could own the data, but not BoundTexture - but that is how it works now.